packages/whenable/src/watch.js

```diff
@@ -0,0 +1,193 @@
+// @ts-check
+/* global globalThis */
+import { E } from '@endo/far';
+import { M } from '@endo/patterns';
+
+import { prepareWhenableKits } from './whenable.js';
+import { getWhenable0, getFirstWhenable } from './whenable-utils.js';
+
+const { Fail } = assert;
+
+export const PromiseWatcherI = M.interface('PromiseWatcher', {
+  onFulfilled: M.call(M.any()).rest(M.any()).returns(M.any()),
+  onRejected: M.call(M.any()).rest(M.any()).returns(M.any()),
+});
+
+/**
+ * @typedef {object} PromiseWatcher
+ * @property {(...args: unknown[]) => void} [onFulfilled]
+ * @property {(...args: unknown[]) => void} [onRejected]
+ */
+
+/** @type {(p: PromiseLike<any>, watcher: PromiseWatcher, ...args: unknown[]) => void} */
+let watchPromise = /** @type {any} */ (globalThis).VatData?.watchPromise;
```

Comment:
Aha! An agoric-sdk dependency invisible from package.json dependencies. I understand why. No change suggested. 

But should this be in `VatData`, or is it more like optionally endowing a `passStyleOf`, which originally was also `VatData` but changed to `globalThis[PassStyleOfEndowmentSymbol]` https://github.com/endojs/endo/blob/1ff3c4b45a4f9f6bf723d3200548db37df46989f/packages/pass-style/src/passStyleOf.js#L193

Attn @mhofman

Reply:
This was dirty.  If I separate the layering and make `@endo/whenable`, I'd provide `watchPromise` as a power to the `prepareWhenableModule` function, and supply the `globalThis.VatData?.watchPromise` default in `@agoric/internal`.

Reply:
This has me wondering: Should zones be able to abstract over this promise-ish API, so that we have one adapted to each zone? If so, should the heap version move to endo when base-zone moves to endo? Should this then be in base-zone rather than @agoric/internal?

Reply:
This is an interesting question, but I'm deliberately deferring it (and other close integrations with our platform) in order to accomplish just the base user-level abstractions needed to express long-lived promises that tolerate upgrades of vats holding either the settler or the whenable.

Reply:
Let's keep deferring the migrate-ability until @agoric/base-zone is about to migrate.

Once we have `E.watch`, we'd never need `watchPromise`, right? It would just be deprecated legacy in favor of `E.watch`?

Reply:
Not "just" deprecated legacy, because `E.watch` is built from `watchPromise`, if provided by liveslots. But aside from than that internal layering as well as legacy uses, would there still be any reason for normal developers to be aware of `watchPromise`?

Reply:
It makes sense to me that watchPromise would not generally appear in user-level code.

Reply:
Good. Resolving.